src/test-apps/TapAddrAutoconf.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TapAddrAutoconf.h"
+#include <inttypes.h>
+#include <stdint.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <ifaddrs.h>
+#include <arpa/inet.h>
+
+int CollectTapAddresses(std::vector<char *>  &addresses, const char * ifName) {
+    struct ifaddrs *addrsList, *curAddr;
+    const int err = getifaddrs(&addrsList);
+
+    if (err == 0)
+    {
+        curAddr = addrsList;
+        while (curAddr)
+        {
+            if (strcmp(ifName, curAddr->ifa_name) == 0)
+            {
+                char buf[128];
```

Comment:
128 -> Magic Number? This should probably be INET6_ADDRSTRLEN.

Reply:
done